lighthouse-core/lib/errors.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const strings = require('./strings');
+
+class LighthouseError extends Error {
+  constructor(errorDefinition) {
+    super(errorDefinition.code);
+    Error.captureStackTrace(this, LighthouseError);
+    this.friendlyMessage = errorDefinition.message;
+  }
+}
+
+const errors = {
+  NO_SPEEDLINE_FRAMES: {message: strings.didntCollectScreenshots},
+  SPEEDINDEX_OF_ZERO: {message: strings.didntCollectScreenshots},
+  NO_SCREENSHOTS: {message: strings.didntCollectScreenshots},
```

Comment:
just to note, this one isn't used yet. we created it for https://github.com/paulirish/speedline/blob/53b51d283fcbed69710eda58e8c3539f0fb9134d/src/frame.js#L105

do you want to catch the provided error and rethrow as LHError?

Reply:
done